code/modules/mob/living/simple_animal/hostile/hivebots/hivebot_beacon.dm

```diff
@@ -34,40 +34,110 @@
 	minbodytemp = 0
 	speed = -10
 	destroy_surroundings = 0
-	var/bot_type
-	var/bot_amt = 160 //Number of total bots that are spawned before the beacon disappears completely.
-	var/max_bots = 40 //Number of bots linked to this beacon specifically that can exist, before spawning more is halted.
-	var/list/linked_bots = list()
+	attack_emote = "focuses on"
+	psi_pingable = FALSE
+
+	/**
+	 * Number of total bots that are spawned before the beacon disappears completely
+	 *
+	 * This gets updated in /Initialize() with a formula based on the
+	 * `total_hivebots_to_spawn_to_playing_players_scaling_factor` variable
+	 */
+	var/total_hivebots_to_spawn = 160
+
+	/**
+	 * The scaling factor for the number of total hivebots to spawn respective to playing players
+	 *
+	 * Eg: scaling factor of 1 would mean that the beacon would spawn 1 additional hivebot per player
+	 */
+	var/total_hivebots_to_spawn_to_playing_players_scaling_factor = 1
```

Comment:
The name isn't too great because the use of "scaling factor" implies that this is a multiplication, when it isnt.

This should either be turned into a factor or updated so the name doesn't imply it's a factor when it isn't.

Reply:
How would you call it?